### DISCUSSION TITLE
How can I fix Timestamp.toDate() returning "undefined is not an object" even when I explicitly define to only check for that value if the Timestamp isn't null?

### DISCUSSION THREAD

**Discussion by faz919** [invertase/react-native-firebase#6032] [Q&A] [answered]
I've assigned a "time" value to each document that I upload to Firestore that runs ```time: firestore.Timestamp.fromDate(new Date())```. Then, later on in my code, I need to call on this time value to set text that says ```moment(selectedImage.time.toDate()).fromNow()```. If I've already specified a "selectedImage" and thus retrieved a time value, then I get no errors from the console. However, on app start, there is no selectedImage value and therefore the time and time.toDate() values are null, and I get an error returned that can't be dismissed/ignored. So I've already tried specifying to only retrieve the ```selectedImage.time.toDate()``` value if that value or any of its parent values aren't null, but it always checks for time.toDate() even if it's not supposed to and returns an error that doesn't let me use my app. How can I fix this problem?

(I can get the functionality working fine if I remove the .toDate() function, load up the app, select an image, and then add back the .toDate() function and reload, but if I tell the program to do that same thing automatically by not calling upon the toDate() function until selectedImage.time is specified, it still checks for toDate() and returns an error.)
![Screenshot_1643185557](https://user-images.githubusercontent.com/63302288/151128251-2e989c49-0754-43a2-9e8e-4a1d72967627.png)

![Screenshot_1643185504](https://user-images.githubusercontent.com/63302288/151128129-8698f1f0-0311-42dd-801c-b320204b5d46.png)

(This also doesn't work, still checks for .toDate() being undefined on app startup even if it's explicitly told not to do so)
![image](https://user-images.githubusercontent.com/63302288/151128584-034f35ed-bf1e-46ca-800a-972047d1d5c4.png)

**Comment by andersonaddo** [comment, marked as answer]
>  However, on app start, there is no selectedImage value and therefore the time and time.toDate() values are null


If what you're saying is true, selectedImage is undefined and time and  time.toDate() would be erroneous
`null` and `undefined` are very different.

However, your error says something else. It's complaining that time is undefined, not selectedImage. It looks like you're not properly understanding the data flow of your app.
